pkg/vfs/download_store.go

```diff
@@ -0,0 +1,144 @@
+package vfs
+
+import (
+	"encoding/hex"
+	"sync"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/crypto"
+)
+
+// A DownloadStore is essentially an object to store Archives & Files by keys
+type DownloadStore interface {
+	AddFile(f string) (string, error)
+	AddArchive(a *Archive) (string, error)
+	GetFile(k string) (string, error)
+	GetArchive(k string) (*Archive, error)
+}
+
+type fileRef struct {
+	Path      string
+	ExpiresAt time.Time
+}
+
+// downloadStoreTTL is the time an Archive stay alive
+const downloadStoreTTL = 1 * time.Hour
+const downloadStoreCleanInterval = 1 * time.Hour
+
+var storeStoreMutex = &sync.Mutex{}
+var storeStore map[string]*memStore
+
+func init() {
+	go cleanDownloadStoreInterval()
+}
+
+func cleanDownloadStoreInterval() {
+	for range time.Tick(downloadStoreCleanInterval) {
+		cleanDownloadStore()
+	}
+}
+
+func cleanStore(s *memStore) {
+	now := time.Now()
+	for k, f := range s.Files {
+		if now.After(f.ExpiresAt) {
+			delete(s.Files, k)
+		}
+	}
+	for k, a := range s.Archives {
+		if now.After(a.ExpiresAt) {
+			delete(s.Archives, k)
+		}
+	}
+}
+
+func cleanDownloadStore() {
+	storeStoreMutex.Lock()
```

Comment:
would use a `defer Unlock()` on all locking methods (even the correct ones), for clarity